### DISCUSSION TITLE
I just wanted to make some changes  to agent dialog boxes.

### DISCUSSION THREAD

**Discussion by a-imran99** [Ylianst/MeshCentral#5206] [Q&A] [answered]
hi @Ylianst @si458 @krayon007,

I wanted to make some changes to the User-consent dialog box and some other dialog boxes, Thanks for the configuration that you have provided. The changes I need to make are about layout and remove some buttons from dialog boxes.

I have tried building the agent from scratch by building the MeshAgent code in visual studio. 
I have altered the modules folder where the javascript code for dialog boxes are written. But the changes I made are not reflected

Then I came to realize that all the JS files are somehow converted to base64 in iLibDuktape_Polyfills.c and stored predefined.

![image](https://github.com/Ylianst/MeshCentral/assets/77147011/aba4c926-486a-4063-9a2c-43f04f3d719e)

How can I make changes and replace it here in this file  iLibDuktape_Polyfills.c 

Help would be thankful :)

**Comment by si458** [comment]
@a-imran99 did you ever work out how the polyfills.c work?
i have made a few changes as well and then realised i have to polyfill it also
EDIT: i dont think its base64 encoded as if you decode it manually it doesnt convert to the JS file ?

**Comment by si458** [comment]
ok update! i think ive worked out how to do it!

`.\MeshService64.exe -import --expandedPath=C:\Users\ASUS\Documents\GitHub\MeshAgent\modules --filePath=C:\Users\ASUS\Documents\GitHub\MeshAgent\microscript\ILibDuktape_Polyfills.c --modulesPath=C:\Users\ASUS\Documents\GitHub\MeshAgent\modules`

this will import ALL js modules inside `modules` into the polyfills!
but oversally its not quite right as we dont want to import every single JS file!
so just need to work out how to import a few modules and not every single one

**Comment by si458** [comment, marked as answer]
ok ive found what to do again!
1. copy the meshservice64.exe to the root of your project.
2. extract the embedded javascripts `MeshService64.exe -export`
3. step 2 will create a new folder called `modules_expanded`, edit the modules in there and NOT `modules`
4. build the polyfill `MeshService64.exe -import --filePath=.\microscript\ILibDuktape_Polyfills.c --modulesPath=.\modules`
5. build the exe in visual studio
6. copy `meshservice64.exe` to `agents`
7. restart meshcentral, away it goes!!!
